java.hints/src/org/netbeans/modules/java/hints/suggestions/ConvertToVarHint.java

```diff
@@ -0,0 +1,449 @@
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.MSG_ConvertibleToVarType(), fix);
+
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription if variable initializer is a Literal
+     */
+    @Hint(displayName = "#DN_CanUseVarForLiteralRef", description = "#DESC_CanUseVarForLiteralRef", category = "suggestions")
+    @TriggerPattern("$mods$ $type $var = $value;") //NOI18N
+
+    public static ErrorDescription checkLiteralInit(HintContext ctx) {
+
+        TreePath initPath = ctx.getVariables().get("$value");
+
+        if (initPath == null) {
+            return null;
+        }
+
+        Tree l = initPath.getLeaf();
+
+        // checks rhs is a literal or not
+        if (!(isLiteral(l))) {
+            return null;
+        }
+
+        if (!isHintEnabled(ctx)) {
+            return null;
+        }
+
+        Fix fix = new FixImpl(ctx.getInfo(), ctx.getPath()).toEditorFix();
+        return ErrorDescriptionFactory.forName(ctx, ctx.getPath(), Bundle.MSG_ConvertibleToVarType(), fix);
+
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription if variable initializer is Lambda expression.
+     */
+    @Hint(displayName = "#DN_CanUseVarForLambdaExpRef", description = "#DESC_CanUseVarForLambdaExpRef", category = "suggestions")
+    @TriggerTreeKind(Kind.LAMBDA_EXPRESSION)
+    public static ErrorDescription checkLambdaExpr(HintContext ctx) {
+        TypeMirror samType = ctx.getInfo().getTrees().getTypeMirror(ctx.getPath());
+
+        if (!isHintEnabled(ctx)) {
+            return null;
+        }
+
+        return ErrorDescriptionFactory.forTree(ctx, ctx.getPath(), Bundle.MSG_ConvertibleToVarType(), new LambdaExprFixImpl(ctx.getInfo(), ctx.getPath()).toEditorFix());
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription if variable initializer is Anonymous Object.
+     */
+    @Hint(displayName = "#DN_CanUseVarForAnonymousObjRef", description = "#DESC_CanUseVarForAnonymousObjRef", category = "suggestions")
+
+    @TriggerPattern("new $clazz($params$) { $method; }")  //NOI18N
+
+    public static ErrorDescription checkAnonymousObj(HintContext ctx) {
+
+        if (!isHintEnabled(ctx)) {
+            return null;
+        }
+
+        return ErrorDescriptionFactory.forTree(ctx, ctx.getPath(), Bundle.MSG_ConvertibleToVarType(), new FixImpl(ctx.getInfo(), ctx.getPath()).toEditorFix());
+    }
+
+    /**
+     *
+     * @param ctx : HintContext
+     * @return ErrorDescription if variable initializer Object uses diamond
+     * operator(<>).
+     */
+    @Hint(displayName = "#DN_CanUseVarForDiamondInterfaceRef", description = "#DESC_CanUseVarForDiamondInterfaceRef", category = "suggestions")
+    @TriggerPattern("new $type<$T$>($params$)") //NOI18N
+    public static ErrorDescription checkdiamondType(HintContext ctx) {
+
+        TreePath treePath = ctx.getPath();
+
+        if (!isHintEnabled(ctx)) {
```

Comment:
and here and the next one.

Reply:
this is addressed in commit : https://github.com/apache/incubator-netbeans/pull/463/commits/b733846a4ab3081e2b4d56531ab3c3d041fdd93e